client/src/services/agenda.ts

```diff
@@ -0,0 +1,81 @@
+import { socket } from "@/socket";
+import { create } from "zustand";
+import { useEffect } from "react";
+import {
+  OngoingAgenda,
+  TerminatedAgenda,
+  Agenda,
+} from "biseo-interface/agenda";
+
+interface Voted {
+  id: number;
+  voters: {
+    voted: number;
+    total: number;
+  };
+}
+
+interface Reminded {
+  agendaId: number;
+  message: string;
+}
+
+interface AgendaState {
+  agendas: Agenda[];
+  voteAgenda: (choiceId: number) => void;
+  retrieveAgendas: () => void;
+}
+
+const useAgenda = create<AgendaState>((set, get) => ({
+  agendas: [],
+  voteAgenda: async (choiceId) => {
+    try {
+      await socket.emitAsync("agenda.vote", { choiceId });
+    } catch (error) {
+      // TODO: handle error
+      console.error("Vote error:", error);
+    }
+  },
+  retrieveAgendas: async () => {
+    try {
+      const agendas = await socket.emitAsync("agenda.retrieveAll", {
+        lastAgendaId: get().agendas[0]?.id || null,
+      });
+      set((state) => ({ agendas: [...agendas, ...state.agendas] }));
+    } catch (error) {
+      // TODO: handle error
+      console.error("Retrieve Agendas error:", error);
+    }
+  },
+}));
+
+socket.on("agenda.started", (ongoingAgenda: OngoingAgenda) => {
+  useAgenda.setState((state) => ({
+    agendas: [...state.agendas, ongoingAgenda],
+  }));
+});
```

Comment:
여기에서 `socket`이 `BiseoSocket` 타입을 가지고 있는데, 여기에서 각 이벤트에 대한 핸들러 타입을 모두 가지고 있기 때문에 핸들러의 변수인 `ongoingAgenda` 또한 명시적으로 지정해 주지 않아도 자동으로 infer 됩니다. 

```suggestion
socket.on("agenda.started", (ongoingAgenda) => {
  useAgenda.setState((state) => ({
    agendas: [...state.agendas, ongoingAgenda],
  }));
});
```

이와 같이 implicit하게 타입을 정의할 수 있는 경우에는 보통 이를 명시적으로 나타내지 않는 것이 best practice로 취급되고, 또한 추후 API 변경에도 유연하게 대응할 수 있기 때문에 명시하지 않는 것이 맞는 것 같습니다. 

관련 typescript type inference 관련해서 읽어보면 좋을 것 같은 자료입니다. 
- https://www.typescriptlang.org/docs/handbook/type-inference.html